internal/jobs/webscraper.go

```diff
@@ -46,14 +46,25 @@ func (ws *WebScraper) GetStructuredCapabilities() teetypes.WorkerCapabilities {
 func (ws *WebScraper) ExecuteJob(j types.Job) (types.JobResult, error) {
 	logrus.Info("Starting ExecuteJob for web scraper")
 
-	// Step 1: Unmarshal arguments
-	args := &teeargs.WebSearchArguments{}
-	logrus.Info("Unmarshaling job arguments")
-	if err := j.Arguments.Unmarshal(args); err != nil {
+	// Step 1: Use centralized type-safe unmarshaller
+	jobArgs, err := teeargs.UnmarshalJobArguments(teetypes.JobType(j.Type), map[string]any(j.Arguments))
+	if err != nil {
 		logrus.Errorf("Failed to unmarshal job arguments: %v", err)
-		return types.JobResult{Error: fmt.Sprintf("Invalid arguments: %v", err)}, err
+		ws.stats.Add(j.WorkerID, stats.WebInvalid, 1)
+		return types.JobResult{Error: fmt.Sprintf("Invalid arguments: %v", err)}, nil
+	}
+
+	// Type assert to Web arguments
+	webArgs, ok := teeargs.AsWebArguments(jobArgs)
+	if !ok {
+		logrus.Errorf("Expected Web arguments for job ID %s, type %s", j.UUID, j.Type)
+		ws.stats.Add(j.WorkerID, stats.WebInvalid, 1)
+		return types.JobResult{Error: "invalid argument type for Web job"}, nil
 	}
-	logrus.Infof("Job arguments unmarshaled successfully: %+v", args)
+
+	// Convert to the concrete type for easier access
+	args := webArgs.(*teeargs.WebSearchArguments)
```

Comment:
ditto, why do we need this cast? Can't `teeargs.AsWebArguments` return the proper type?